src/Functions/distribution.cpp

```diff
@@ -0,0 +1,452 @@
+    static constexpr size_t getNumberOfArguments() { return 2; }
+
+    void generate(Float64 d1, Float64 d2, ColumnFloat64::Container & container) const
+    {
+        auto distribution = std::fisher_f_distribution<>(d1, d2);
+        for (auto & elem : container)
+            elem = distribution(thread_local_rng);
+    }
+};
+
+struct BernoulliDistribution
+{
+    static constexpr const char * getName() { return "bernoulliDistribution"; }
+    static constexpr size_t getNumberOfArguments() { return 1; }
+
+    void generate(Float64 p, ColumnUInt8::Container & container) const
+    {
+        if (p < 0.0f || p > 1.0f)
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Argument of function {} should be inside [0, 1] because it is a probability", getName());
+
+        auto distribution = std::bernoulli_distribution(p);
+        for (auto & elem : container)
+            elem = static_cast<UInt8>(distribution(thread_local_rng));
+    }
+};
+
+struct BinomialDistribution
+{
+    static constexpr const char * getName() { return "binomialDistribution"; }
+    static constexpr size_t getNumberOfArguments() { return 2; }
+
+    void generate(UInt64 t, Float64 p, ColumnUInt64::Container & container) const
+    {
+        if (p < 0.0f || p > 1.0f)
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Argument of function {} should be inside [0, 1] because it is a probability", getName());
+
+        auto distribution = std::binomial_distribution(t, p);
+        for (auto & elem : container)
+            elem = static_cast<UInt64>(distribution(thread_local_rng));
+    }
+};
+
+struct NegativeBinomialDistribution
+{
+    static constexpr const char * getName() { return "negativeBinomialDistribution"; }
+    static constexpr size_t getNumberOfArguments() { return 2; }
+
+    void generate(UInt64 t, Float64 p, ColumnUInt64::Container & container) const
+    {
+        if (p < 0.0f || p > 1.0f)
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Argument of function {} should be inside [0, 1] because it is a probability", getName());
+
+        auto distribution = std::negative_binomial_distribution(t, p);
+        for (auto & elem : container)
+            elem = static_cast<UInt64>(distribution(thread_local_rng));
+    }
+};
+
+struct PoissonDistribution
+{
+    static constexpr const char * getName() { return "poissonDistribution"; }
+    static constexpr size_t getNumberOfArguments() { return 1; }
+
+    void generate(UInt64 n, ColumnUInt64::Container & container) const
+    {
+        auto distribution = std::poisson_distribution(n);
+        for (auto & elem : container)
+            elem = static_cast<UInt64>(distribution(thread_local_rng));
+    }
+};
+
+}
+/// Function which will generate values according to the distribution
+/// Accepts only constant arguments
+template <typename Distribution>
+class FunctionDistribution : public IFunction
+{
+private:
+    mutable Distribution distribution;
```

Comment:
btw, why is this mutable?

Reply:
Just some leftover from previous version of my code where Distribution had a non const method, but `executeImpl` in function was const